python/paddle/fluid/contrib/tests/test_bf16_utils.py

```diff
@@ -0,0 +1,194 @@
+            custom_fp32_list={'elementwise_add'})
+        self.assertEqual(amp_lists_.bf16_list, bf16_list)
+        self.assertEqual(amp_lists_.fp32_list, fp32_list)
+        self.assertEqual(amp_lists_.gray_list, gray_list)
+
+    def test_amp_lists_5(self):
+        bf16_list = copy.copy(amp.bf16.amp_lists.bf16_list)
+        fp32_list = copy.copy(amp.bf16.amp_lists.fp32_list)
+        gray_list = copy.copy(amp.bf16.amp_lists.gray_list)
+
+        # 5. w=None, b={'elementwise_add'}
+        fp32_list.add('elementwise_add')
+        bf16_list.remove('elementwise_add')
+
+        amp_lists_ = amp.AutoMixedPrecisionListsBF16(
+            custom_fp32_list={'elementwise_add'})
+        self.assertEqual(amp_lists_.bf16_list, bf16_list)
+        self.assertEqual(amp_lists_.fp32_list, fp32_list)
+        self.assertEqual(amp_lists_.gray_list, gray_list)
+
+    def test_amp_lists_6(self):
+        bf16_list = copy.copy(amp.bf16.amp_lists.bf16_list)
+        fp32_list = copy.copy(amp.bf16.amp_lists.fp32_list)
+        gray_list = copy.copy(amp.bf16.amp_lists.gray_list)
+
+        # 6. w=None, b={'lstm'}
+        fp32_list.add('lstm')
+
+        amp_lists_ = amp.AutoMixedPrecisionListsBF16(custom_fp32_list={'lstm'})
+        self.assertEqual(amp_lists_.bf16_list, bf16_list)
+        self.assertEqual(amp_lists_.fp32_list, fp32_list)
+        self.assertEqual(amp_lists_.gray_list, gray_list)
+
+    def test_amp_lists_7(self):
+        # 7. w={'lstm'} b={'lstm'}
+        # raise ValueError
+        self.assertRaises(ValueError, amp.AutoMixedPrecisionListsBF16,
+                          {'lstm'}, {'lstm'})
+
+    def test_amp_lists_8(self):
+        bf16_list = copy.copy(amp.bf16.amp_lists.bf16_list)
+        fp32_list = copy.copy(amp.bf16.amp_lists.fp32_list)
+        gray_list = copy.copy(amp.bf16.amp_lists.gray_list)
+
+        fp32_list.add('reshape2')
+        gray_list.remove('reshape2')
+
+        amp_lists_ = amp.AutoMixedPrecisionListsBF16(
+            custom_fp32_list={'reshape2'})
+        self.assertEqual(amp_lists_.bf16_list, bf16_list)
+        self.assertEqual(amp_lists_.fp32_list, fp32_list)
+        self.assertEqual(amp_lists_.gray_list, gray_list)
+
+    def test_amp_list_9(self):
+        bf16_list = copy.copy(amp.bf16.amp_lists.bf16_list)
+        fp32_list = copy.copy(amp.bf16.amp_lists.fp32_list)
+        gray_list = copy.copy(amp.bf16.amp_lists.gray_list)
+
+        bf16_list.add('reshape2')
+        gray_list.remove('reshape2')
+
+        amp_lists_ = amp.AutoMixedPrecisionListsBF16(
+            custom_bf16_list={'reshape2'})
+        self.assertEqual(amp_lists_.bf16_list, bf16_list)
+        self.assertEqual(amp_lists_.fp32_list, fp32_list)
+        self.assertEqual(amp_lists_.gray_list, gray_list)
+
+    def test_find_op_index(self):
+        block = fluid.default_main_program().global_block()
+        op_desc = core.OpDesc()
+        idx = amp.bf16.amp_utils.find_op_index(block.desc, op_desc)
+        assert (idx == -1)
```

Comment:
I don't see any function named `find_op_index` inside [`amp_utils.py`](https://github.com/PaddlePaddle/Paddle/pull/31093/files#diff-8d2f4bbd27be57cefa65ae19ec4b6941f6270fec4dc16fa293c3f52955b5e44d) file. So if you didn't modified the behavior of this function is this test really needed? There is already a test for it [this one](https://github.com/PaddlePaddle/Paddle/blob/d8f810cc161b837464f529fac064a2093700c89f/python/paddle/fluid/contrib/tests/test_fp16_utils.py#L25-L29) for `fp16`.

Reply:
Well it's defined in `amp_utils.py` as an import from `fp16_utils.py` and that way can be used as defined here. Maybe that test is done a bit in advance but we'll gonna use the exactly the same `find_op_index` in bf16 future so I didn;t want to to copy-paste it here.